geps/gep-2198.md

```diff
@@ -0,0 +1,105 @@
+
+- the token content is tested for the existence of some claim property or value
+- claim values may be extracted and saved as header value(s) to the request
+- the entire token or a sub-set of claims may be forwarded directly to backend services for evaluation
+
+## Persona(s)
+
+The Personas that can be involved in managing JWT Policy can be:
+
+- platform / cluster operations
+- application developer / admins
+- user account manager
+- security operations
+
+## Prior related art
+
+Here are some samples of JWT implementations in a similar problem space.  Each implementation differs in terminology and implementation.
+
+- NGINX Ingress Controller:
+
+  - [https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-local-kubernetes-secret](https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-local-kubernetes-secret)
+  - [https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-jwks-from-remote-location](https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#jwt-using-jwks-from-remote-location)
+  
+- Kong:
+
+  - [https://docs.konghq.com/hub/kong-inc/jwt/configuration/](https://docs.konghq.com/hub/kong-inc/jwt/configuration/)
+
+- Envoy Gateway:
+
+  - [https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider](https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider)
+
+- Istio:
+
+  - [https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/](https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/jwt-route/](https://istio.io/latest/docs/tasks/security/authentication/jwt-route/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/](https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/)
+
+## Properties of a JWT Validation Policy
+
+The validation policy is two fold, it establishes the relationship with the Token and how to validate it.
+According to [https://datatracker.ietf.org/doc/html/rfc7519#section-4.1](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) claims cannot be assumed mandatory.  At a minimum a claim should be able to be validated following the process outlined here which only requires a method to decrypt if necessary: [https://datatracker.ietf.org/doc/html/rfc7519#section-7.2](https://datatracker.ietf.org/doc/html/rfc7519#section-7.2)
+
+The statements made above point to the primary points of divergence between the JWT validation implementations in the prior art section.
+
+Some very high level proposed properties to open the conversation based on comparing the implementations and reviewing the RFC.
+
+- name / realm (required) - the name / realm of the JWT, this is commonly a string representing a specific application.  It is seen used in response and log messages.
```

Comment:
Not sure why name/realm would be required. Log messages are important but not that important. 

Issuer field is probably more useful as a key. 

For jwksURI - it should be mentioned 'if missing, implementation should fetch using OIDC'. And perhaps have a way to specify the public keys directly. 

Not sure if keyCache is that important ( implementation detail ), and 'token' is a very bad name ( it usually means a secret ). If you mean the 'header' or 'parameter' - better be explicit.